lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -1240,6 +1245,26 @@ ChangeResult TypeAnalyzer::visitAtenTransposeIntOp(
   return getLatticeElement(op.getResult()).join(knowledge);
 }
 
+ChangeResult TypeAnalyzer::visitAtenTOp(
+	AtenTOp op,
+	ArrayRef<LatticeElement<ValueKnowledge> *> operands) {
+	auto input = operands[0]->getValue();
+	auto knowledge =
+		ValueKnowledge::getNotNonePessimisticValueState(op.getContext());
+	knowledge.dtype = input.dtype;
+	knowledge.hasSizes = input.hasSizes;
+	knowledge.sizes = input.sizes;
+	int64_t inputRank = input.sizes.size();
+	if (inputRank>2) {
+		std::string error_message = "t() expects a tensor with <=2 dimensions, but self is " + std::to_string(inputRank) + "D";
+		llvm::report_fatal_error(error_message.c_str());
+	}
+	if (inputRank == 2) {
+		std::swap(knowledge.sizes[0], knowledge.sizes[1]);
```

Comment:
This will give an error if dynamic dimensions are passed.

Reply:
Ya it passes fine in dynamic as well. Do I need to change something here?

Reply:
It looks fine to me. if the input rank is 2, then there has to be enough elements in the array to perform the swap without any issues.